### DISCUSSION TITLE
Uc mode with manual usage

### DISCUSSION THREAD

**Discussion by ggforces** [seleniumbase/SeleniumBase#2273] [Q&A] [answered]
How can I perform all clicks manually using 'sb.driver.uc_click' in SeleniumBase? In other words, instead of automation, the user will click manually. Will this method make it undetected?

**Comment by STARKILLER-1** [comment]
I think not, although I'm not sure, uc_click disconnects and reconnects to the page during a click, whereas a normal user click does not do this. (https://youtu.be/5dMFI3e85ig?si=k9tcX6N7Wn4Cpz2m 26:43 in guide video)

**Comment by mdmintz** [comment, marked as answer]
Regular manual clicking during UC Mode can get you detected because those websites aren't looking specifically at the click, but they are looking at your environment settings / Chrome variables / etc. The `driver.uc_click(selector)` method disconnects chromedriver from Chrome immediately after the click for a very short time... If a website looks for traces of chromedriver, then it won't find it if chromedriver disconnected fast enough.

However, if you want to do some manual actions during UC Mode, first call `driver.reconnect(timeout)`, which will disconnect chromedriver from Chrome for that many seconds, and then you can perform manual actions safely before your time runs out.

**Reply by ggforces** [reply]
For example, open example.com, have all clicks on example.com go undetected. The crucial part is not just mimicking driver.uc_click but ensuring that it remains undetectable. Is this possible? Instead of a few manual steps, how should I proceed to have the user perform all the actions and avoid detection?

**Reply by mdmintz** [reply]
If you have lots of manual actions that you want to do undetected, then set a large `timeout` when you call `driver.reconnect(timeout)`, and then do all your actions before your time runs out. If you need more time, make sure you set a larger timeout in advance.

**Reply by ggforces** [reply]
I would like to ask a few questions because I haven't fully grasped the function of Driver.reconnect. For example, if I set 9999 seconds for the timeout, is there a way to display this as a countdown for the user? Also, does the reconnect functionality only apply to a single page? If we initiate a reconnect to page x and the user opens a new tab, navigates to page y, can we assume that actions on page y are undetected, but when the user returns to page x, actions on page x remain undetected? Or does opening page y eliminate the undetected status? My last question is, if we continuously use a while true loop to reconnect with a 1-second timeout for the current link, how does it exactly conclude? Can it consistently make every opened link undetected?

**Reply by mdmintz** [reply]
That's a lot of questions about a method originally from:
https://github.com/ultrafunkamsterdam/undetected-chromedriver/blob/cea80717c5a3d95ccf5c40e6e38081d5454ec7a5/undetected_chromedriver/__init__.py#L705
Other than fixes, existing UC Mode methods remain as they were to preserve functionality and reduce confusion for people crossing over between frameworks.
For questions about specifics on `reconnect()`, ask the original author.
